executor/src/witgen/mod.rs

```diff
@@ -398,9 +422,26 @@ impl<'a, T: FieldElement> FixedData<'a, T> {
         }
     }
 
+    pub fn filter_identities(self, f: impl Fn(&FixedData<T>, &Identity<T>) -> bool) -> Self {
+        let keep = self
+            .identities
+            .iter()
+            .map(|id| f(&self, id))
+            .collect::<Vec<_>>();
+        let identities = self
+            .identities
+            .into_iter()
+            .zip_eq(keep)
+            .filter(|(_, keep)| *keep)
+            .map(|(id, _)| id)
+            .collect();
+        Self { identities, ..self }
+    }
+
     pub fn with_global_range_constraints(
         self,
         global_range_constraints: GlobalConstraints<T>,
+        retained_identities: Vec<Identity<T>>,
```

Comment:
I think the fact that this swaps out the identities warrants a change of function name.

Reply:
Hmm, I didn't find a good name, and maybe it's not good to have a function that does two things anyway. I undid the change and used `filter_identities` instead.